### PR TITLE
Add similarity tab and attendance to similar schools

### PR DESCRIPTION
**Pull request by vipinsamreddy**
## Description

Pr Include changes to add similarity tab and attendance tab to the similar schools for primary 

## Related Trello Ticket
https://trello.com/c/oyli7KKH/899-ps45-view-similar-schools-similarity-tab-school-characteristics-table-and-display-school-value
https://trello.com/c/hVe2o30m/1065-ps74-view-similar-schools-attendance-page

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit tests added/updated 
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [ ] Tested on [Environment: Dev/Staging/Local]

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have verified accessibility requirements are met

## Screenshots (if applicable)

Add UI changes, logs, console outputs anything to help reviewers.

### CODE REVIEW

File: SAPSec.Core/Features/Primary/PrimaryAttendanceComparisonDataProvider.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+using SAPSec.Core.Features.SimilarSchools;
+using SAPSec.Data.Repositories;
+
+namespace SAPSec.Core.Features.Primary;
+
+public class PrimaryAttendanceComparisonDataProvider(
```
Comment (stusherwin) [inline SAPSec.Core/Features/Primary/PrimaryAttendanceComparisonDataProvider.cs]: Nitpick but can we use `Absence` rather than `Attendance` in the back-end when referring to absence data? It is technically absence data after all, it's only called Attendance in the UI

File: SAPSec.Core/Features/SimilarSchools/UseCases/GetPrimaryCharacteristicsComparison.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    public required PrimarySimilarSchoolCharacteristicValue<int> PupilCount { get; init; }
+    public required PrimarySimilarSchoolCharacteristicValue<decimal> PupilStabilityRate { get; init; }
+    public required PrimarySimilarSchoolCharacteristicValue<decimal> AverageIdaciScore { get; init; }
+    public required PrimarySimilarSchoolCharacteristicValue<decimal> PupilsWithSenSupportPercentage { get; init; }
+    public required PrimarySimilarSchoolCharacteristicValue<decimal> PupilsWithEhcPlanPercentage { get; init; }
+}
+
+public record PrimarySimilarSchoolCharacteristicValue<T>(
+    T CurrentSchoolValue,
+    T SimilarSchoolValue);
```
Comment (stusherwin) [inline SAPSec.Core/Features/SimilarSchools/UseCases/GetPrimaryCharacteristicsComparison.cs]: This idea of a data point with a `CurrentSchoolValue` and a `SimilarSchoolValue` is more generic and looks like it can be shared across both primary and secondary - can we call it something like `SchoolComparisonValue` and extract it out as a domain object?

File: SAPSec.Core/Features/SimilarSchools/UseCases/GetPrimaryCharacteristicsComparison.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        Convert.ToInt32(Math.Round(value, MidpointRounding.AwayFromZero));
+
+    private static decimal RoundWholeNumber(decimal value) =>
+        decimal.Round(value, 0, MidpointRounding.AwayFromZero);
+
+    private static decimal RoundToOneDecimalPlace(decimal value) =>
+        decimal.Round(value, 1, MidpointRounding.AwayFromZero);
+
+    private static decimal RoundToThreeDecimalPlaces(decimal value) =>
+        decimal.Round(value, 3, MidpointRounding.AwayFromZero);
```
Comment (stusherwin) [inline SAPSec.Core/Features/SimilarSchools/UseCases/GetPrimaryCharacteristicsComparison.cs]: We have so many implementations of these rounding functions everywhere, and I believe these are duplicated across the primary and secondary use case? Can extract these out into a helper class so they're in one place and not duplicated?

File: Tests/SAPSec.Core.Tests/Features/Primary/GetSchoolAttendanceComparisonUseCaseTests.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        {
+            [Absence.Filters.Type.Key] = type
+        }));
+
+        var series = response.Absence.Series;
+
+        series.First(s => s.SeriesType == MeasureSeriesType.CurrentSchool).Current.Should().Be((decimal?)currentSchool);
+        series.First(s => s.SeriesType == MeasureSeriesType.SimilarSchool).Current.Should().Be((decimal?)similarSchool);
+        series.First(s => s.SeriesType == MeasureSeriesType.EnglandSchoolsAverage).Current.Should().Be((decimal?)england);
+    }
```
Comment (stusherwin) [inline Tests/SAPSec.Core.Tests/Features/Primary/GetSchoolAttendanceComparisonUseCaseTests.cs:277]: We're missing half the tests that we have in `GetSchoolAttendanceMeasuresUseCaseTests` - it should be a case of replicating these

File: Tests/SAPSec.Core.Tests/Features/SimilarSchools/UseCases/GetPrimaryCharacteristicsComparisonTests.cs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        var sut = CreateSut();
+
+        var result = await sut.Execute(new GetPrimaryCharacteristicsComparisonRequest(currentUrn, similarUrn));
+
+        Assert.Equal(101, result.PupilCount.CurrentSchoolValue);
+        Assert.Equal(103, result.PupilCount.SimilarSchoolValue);
+
+        Assert.Equal(1, result.Polar4Quintile.CurrentSchoolValue);
+        Assert.Equal(3, result.Polar4Quintile.SimilarSchoolValue);
+    }
```
Comment (stusherwin) [inline Tests/SAPSec.Core.Tests/Features/SimilarSchools/UseCases/GetPrimaryCharacteristicsComparisonTests.cs:116]: We're missing tests for most of the properties on `SimilarSchoolsPrimaryValuesEntry` - I expect each of the properties to have its own set of tests around precision, rounding etc. We're also missing tests for validation if there are empty/invalid values in the source data. 

This can be all properties in one test with a test for each type of validation, but as it stands if there was a bug in wiring up or parsing any of these missing properties it would not be caught by any tests.

File: Tests/SAPSec.Test.Integration/Tests/Primary/SimilarSchoolsComparisonIntegrationTests.cs [hunk 6]
```diff
@@ -418,4 +420,151 @@ public async Task AchievedHigherStandardGps_TableView_ShouldShowCorrectValues()
...
+    }
+
+    [Fact]
+    public async Task Similarity_LinksToWhatIsASimilarSchoolPage()
+    {
+        var page = await Fixture.RequestPageAsync(
+            Routes.PrimarySchool(PrimarySchoolUrn).SimilarSchoolComparison(SimilarSchoolUrn));
+
+        var link = page.Links.OfType<IHtmlAnchorElement>()
+            .FirstOrDefault(a => a.TrimmedTextContent().Contains("characteristics used to measure similarity"));
```
Comment (stusherwin) [inline Tests/SAPSec.Test.Integration/Tests/Primary/SimilarSchoolsComparisonIntegrationTests.cs]: This is really brittle - it will fail if the link content changes, which is quite likely. Can we target the link more specifically either by narrowing it down to a specific area of the page, or by giving it a test id?

**Review by stusherwin — CHANGES_REQUESTED** [review]

File: SAPSec.Core/Features/Primary/PrimaryAttendanceComparisonDataProvider.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            throw new NotFoundException($"School not found with URN: {similarSchoolUrn}");
+        }
+
+        var attendances = (await absenceRepository.GetByUrnsAsync(urns))
+            .ToDictionary(x => x.Urn, StringComparer.Ordinal);
+
+        var currentSchoolData = new SchoolData<AbsenceData>(
+            schools[currentSchoolUrn],
+            attendances.TryGetValue(currentSchoolUrn, out var currentAttendance) ? currentAttendance : null);
```
Comment (stusherwin) [inline SAPSec.Core/Features/Primary/PrimaryAttendanceComparisonDataProvider.cs]: Again here we're mixing `absence` and `attendance` - better to have consistent naming

**Review by stusherwin — APPROVED** [review]
